### PR TITLE
Spark connector pom fixes

### PR DESCRIPTION
**Pull request by cbalci**
Making some changes to the pom file for the Pinot Spark Connector for reliability.

**Changes**
- Switch from `maven-assembly-plugin` to `maven-shade-plugin` for building the 'Uber jar'. `maven-shade-plugin` provides better control over dependency inclusions in the build output.
- Shade 'guava' and 'google-protobuf' libraries. This is done for preventing against dependency collisions inside the Spark environment [as explained here](https://cloud.google.com/dataproc/docs/guides/manage-spark-dependencies#avoiding_dependency_conflicts).
- Move grpc related dependencies from 'profile' to package level for clarity.

**Testing**
This change is tested in Uber's Spark environment which is fairly complex and had dependency collisions issues with the previous approach (maven-assembly-plugin fat jar)

`bugfix` `dependencies` `cleanup`

### CODE REVIEW

**Comment by cbalci** [comment]
@xiangfu0 can you please take a look? cc @chenboat

File: pinot-connectors/pinot-spark-connector/pom.xml [hunk 1]
```diff
@@ -199,19 +186,25 @@
...
                 <phase>package</phase>
                 <goals>
-                  <goal>single</goal>
+                  <goal>shade</goal>
                 </goals>
+                <configuration>
+                  <relocations>
+                    <relocation>
+                      <pattern>com</pattern>
+                      <shadedPattern>repackaged.com</shadedPattern>
```
Comment (xiangfu0) [inline pinot-connectors/pinot-spark-connector/pom.xml:200]: Consider follow the pattern here for presto, with your shadeBase parameter?
https://github.com/apache/pinot/blob/master/pinot-connectors/prestodb-pinot-dependencies/presto-pinot-driver/pom.xml#L39

**Review by xiangfu0 — APPROVED** [review]
overall lgtm, thanks for fixing this!

**Comment by cbalci** [comment]
Thanks for the review @xiangfu0!
Added a change per your suggestion. Feel free to merge if it looks good, otherwise please let me know.

**Pull request merged**